Signal/src/Models/SignalAttachment.swift

```diff
@@ -543,8 +546,14 @@ class SignalAttachment: NSObject {
                                                                 return attachment
             }
 
-            if jpgImageData.count <= kMaxFileSizeImage {
-                let recompressedAttachment = SignalAttachment(data : jpgImageData, dataUTI: kUTTypeJPEG as String, filename: filename)
+            guard let dataSource = DataSourceValue.dataSource(with:jpgImageData, fileExtension:"jpg") else {
```

Comment:
This is a bit subtle, but `DataSource`'s have a `fileExtension` property (because they may need to be written to disk and their path/url should have the write file extension) whereas `SignalAttachment` has a UTI property.